format.swift

```diff
@@ -0,0 +1,548 @@
+    let arguments = ProcessInfo.processInfo.arguments
+    guard let labelIndex = arguments.firstIndex(of: "--swift-format") else {
+      return nil
+    }
+    let valueIndex = arguments.index(after: labelIndex)
+    if valueIndex == arguments.endIndex {
+      return nil
+    }
+    return arguments[valueIndex]
+  }
+
+  /// Whether to show verbose output of the steps that are being performed.
+  static var verbose: Bool {
+    hasFlag("--verbose")
+  }
+
+  private static func hasFlag(_ flag: String) -> Bool {
+    ProcessInfo.processInfo.arguments.contains(flag)
+  }
+}
+
+struct SwiftSyntaxFormatter {
+  enum Error: Swift.Error, CustomStringConvertible {
+    case swiftFormatNotFound
+    case lintFailed
+
+    var description: String {
+      switch self {
+      case .swiftFormatNotFound:
+        return "The locally built swift-format could not be found"
+      case .lintFailed:
+        return """
+          The swift-syntax repo is not formatted according to the style guides.
+
+          Run the following command to format swift-syntax
+
+            swift-syntax/format.swift
+
+          If the issue persists, try updating swift-format by running
+
+            swift-syntax/format.swift --update
+
+          Should that still fail, fix any remaining issues manually and verify they match the swift-format style using
+
+            swift-syntax/format.swift --lint
+          """
+      }
+    }
+  }
+
+  /// The branch of swift-format to build.
+  let branch: String = "main"
+
+  /// The configuration to build swift-format in.
+  let configuration: String = "release"
+
+  /// Run `git` in the .swift-format-build directory with the provided arguments.
+  func runGitCommand(_ arguments: String...) throws {
+    try ProcessRunner(
+      executableURL: Paths.gitExec,
+      arguments: ["-C", Paths.swiftFormatBuildDir.path] + arguments
+    ).run(
+      verbose: CommandLineArguments.verbose,
+      captureStdout: false,
+      captureStderr: false
+    )
+  }
+
+  /// Run `swift` with for the `.swift-format-build` package with the provided arguments.
+  func runSwiftCommand(_ action: String, _ arguments: String...) throws {
+    try ProcessRunner(
+      executableURL: Paths.swiftExec,
+      arguments: [action, "--package-path", Paths.swiftFormatBuildDir.path] + arguments
+    ).run(
+      verbose: CommandLineArguments.verbose,
+      captureStdout: false,
+      captureStderr: false
+    )
+  }
+
+  /// Ensure that we have an up-to-date checkout of swift-format in `.swift-format-build`.
+  func cloneOrUpdateSwiftFormat() throws {
+    if FileManager.default.fileExists(atPath: Paths.swiftFormatBuildDir.appendingPathComponent(".git").path) {
+      try runGitCommand("checkout", branch)
+      try runGitCommand("pull")
+    } else {
+      try FileManager.default.createDirectory(atPath: Paths.swiftFormatBuildDir.path, withIntermediateDirectories: true)
+      try runGitCommand("clone", "https://github.com/apple/swift-format.git", ".")
```

Comment:
```suggestion
      try runGitCommand("clone", "https://github.com/apple/swift-format.git", Paths.swiftFormatBuildDir.path)
```

Or am I missing something here? Wouldn't need the create directory above either.

Reply:
`runGitCommand` adds an implicit `-C Paths.swiftFormatBuildDir.path` and the `.` is relative to that.

Reply:
Ah I see 👍